Library/Homebrew/dev-cmd/bump.rb

```diff
@@ -221,55 +299,152 @@ def retrieve_pull_requests(formula_or_cask, name, state:, version: nil)
-      current_version = formula_or_cask.stable.version
-      type = :formula
-      version_name = "formula version"
-    else
-      current_version = Version.new(formula_or_cask.version)
-      type = :cask
-      version_name = "cask version   "
+  sig {
+    params(formula_or_cask: T.any(Formula, Cask::Cask), repositories: T::Array[T.untyped], args: T.untyped,
+           name: String).returns VersionBumpInfo
+  }
+  def retrieve_versions_by_arch(formula_or_cask:, repositories:, args:, name:)
+    is_cask_with_blocks = formula_or_cask.is_a?(Cask::Cask) && formula_or_cask.on_system_blocks_exist?
+    type, version_name = formula_or_cask.is_a?(Formula) ? [:formula, "formula version:"] : [:cask, "cask version:  "]
+
+    old_versions = {}
+    new_versions = {}
+
+    repology_latest = repositories.present? ? Repology.latest_version(repositories) : "not found"
+
+    # When blocks are absent, arch is not relevant. For consistency, we simulate the arm architecture.
+    arch_options = is_cask_with_blocks ? [:arm, :intel] : [:arm]
+
+    arch_options.each do |arch|
+      Homebrew::SimulateSystem.with arch: arch do
+        version_key = is_cask_with_blocks ? arch : :general
+
+        # We reload the formula/cask here to ensure we're getting the correct version for the current arch
+        if formula_or_cask.is_a?(Formula)
+          loaded_formula_or_cask = formula_or_cask
+          current_version_value = formula_or_cask.stable.version
+        else
+          loaded_formula_or_cask = Cask::CaskLoader.load(formula_or_cask.sourcefile_path)
+          current_version_value = Version.new(loaded_formula_or_cask.version)
+        end
+
+        livecheck_latest = livecheck_result(loaded_formula_or_cask)
+
+        new_version_value = if livecheck_latest.is_a?(Version) && livecheck_latest >= current_version_value
+          livecheck_latest
+        elsif repology_latest.is_a?(Version) &&
+              repology_latest > current_version_value &&
+              !loaded_formula_or_cask.livecheckable? &&
+              current_version_value != "latest"
+          repology_latest
+        end.presence
+
+        # Store old and new versions
+        old_versions[version_key] = current_version_value
+        new_versions[version_key] = new_version_value
+      end
     end
 
-    livecheck_latest = livecheck_result(formula_or_cask)
+    # If arm and intel versions are identical, as it happens with casks where only the checksums differ,
+    # we consolidate them into a single version.
+    [:arm, :intel].each do |key|
+      if old_versions[:arm] == old_versions[:intel]
+        old_versions[:general] = old_versions[key]
+        old_versions.delete(key)
+      end
 
-    repology_latest = if repositories.present?
-      Repology.latest_version(repositories)
-    else
-      "not found"
+      if new_versions[:arm] == new_versions[:intel]
+        new_versions[:general] = new_versions[key]
+        new_versions.delete(key)
+      end
     end
```

Comment:
This seems overly complicated. I think the following should do the same:

```ruby
old_vesions = { general: old_version[:arm] } if old_versions[:arm] == old_versions[:intel]
new_versions = { general: new_versions[:arm] } if new_versions[:arm] == new_versions[:intel]
```

Reply:
Thanks! I added a presence check on `arm` otherwise it would replace `general` if `arm` and `intel` are `nil`.